### ISSUE TITLE
Implement type annotations

### ISSUE THREAD

**Issue by niconoe** [inbo/crow#49]

**Comment by niconoe** [comment]
Made a quick experiment with TypeScript that looks fantastic, but seems still difficult to use with Vue at this stage. 

We can revisit when Vue 3 is out, and in the meantime looks for a comment-based type annotation system.

**Comment by peterdesmet** [comment]
What do you mean by type annotation?

**Comment by niconoe** [comment]
@peterdesmet: the basic idea is to add specific comments about the type of the objects in a program to avoid bugs and make the code more readable and maintainable. 

The best introductory article I found after a quick search covers Python, but the logic is exactly the same: https://www.codementor.io/@ben_b/why-you-should-use-type-annotations-in-python-some-of-the-time-k8i2uavin

[TypeScript](https://www.typescriptlang.org/) is a very popular way to do that, but it seems to currently bring a few complications with Vue, so I decided to keep things simple and don't use it yet. The situation might change with Vue 3 (it is itself written in TS, and the community is also taking more and more this route).
